src/components/AccountModule/ClientConnectionInfo.js

```diff
@@ -0,0 +1,161 @@
+import React, { useCallback } from 'react'
+import PropTypes from 'prop-types'
+import styled from 'styled-components'
+import { GU, IconCheck, IconCross, Link, textStyle, useTheme } from '@aragon/ui'
+import ClientSyncedInfo from './ClientSyncedInfo'
+import { getAppPath, getPreferencesSearch } from '../../routing'
+import {
+  getConnectionMessage,
+  useConnectionStatusColor,
+  useNetworkConnectionData,
+} from './utils'
+import {
+  CONNECTION_STATUS_ERROR,
+  CONNECTION_STATUS_HEALTHY,
+  CONNECTION_STATUS_WARNING,
+} from './useSyncInfo'
+
+const FlexWrapper = styled.div`
+  display: inline-flex;
+  align-items: center;
+`
+
+function ClientConnectionInfo({
+  connectionStatus,
+  listening,
+  locator,
+  online,
+  syncDelay,
+}) {
+  const theme = useTheme()
+  const { clientNetworkName } = useNetworkConnectionData()
+  const connectionStatusColor = useConnectionStatusColor(
+    listening && online ? connectionStatus : CONNECTION_STATUS_ERROR
+  )
+
+  const Icon =
+    connectionStatus === CONNECTION_STATUS_HEALTHY ? IconCheck : IconCross
+  let connectionMessage = getConnectionMessage(
+    connectionStatus,
+    listening,
+    online,
+    clientNetworkName
+  )
+
+  if (connectionStatus === CONNECTION_STATUS_HEALTHY) {
+    connectionMessage = `Connected to Ethereum ${clientNetworkName} Network`
```

Comment:
It could be, but, looking at how the refactor has been going, I'd say it would be best to see how to break up `useSyncInfo`, and from that information, compose the `connectionMessage` (could be another hook or a function) so that these messages can just be drilled down 🤔, this way the logic is also lifted up.